### ISSUE TITLE
He's not as good as an example

### ISSUE THREAD

**Issue by sunjunlishi** [zzj0311/pytorch-image-inpainting-using-mixed-convolution#2]
My test found that the actual application did not produce very good results. When my original picture is good, the result is good. But the original picture was not good, and his results were terrible.

**Comment by sunjunlishi** [comment]
![053790](https://user-images.githubusercontent.com/11845664/60951751-6337ca80-a32c-11e9-98ad-ede0d694552d.jpg)
![0](https://user-images.githubusercontent.com/11845664/60951765-6b900580-a32c-11e9-91d3-e0ca5737a480.jpg)
![0](https://user-images.githubusercontent.com/11845664/60951771-6e8af600-a32c-11e9-8993-50c1ee913195.jpg)
 1 src file 2 mask file 3  result

**Comment by sunjunlishi** [comment]
![053790](https://user-images.githubusercontent.com/11845664/60951976-e3f6c680-a32c-11e9-96f8-158235625ae6.jpg)
![0](https://user-images.githubusercontent.com/11845664/60951984-e8bb7a80-a32c-11e9-8b54-4314c06ad6d9.jpg)
![0](https://user-images.githubusercontent.com/11845664/60951994-ee18c500-a32c-11e9-8deb-24f8633cd6f5.jpg)
1  src file 2 mask 3result

**Comment by zzj0311** [comment]
To better illustrate your problem, please include your overall settings (which at least should include your hyperparam, your training and testing scheme, and if possible, part of your dataset). If you are using the provided trained model, you should at least fine-tune the model using your own dataset. And, why the groud truth is the same as the input in your result?

If you are more familiar to Chinese, feel free to use that.

**Comment by sunjunlishi** [comment]
因为具体里面的代码我没有完全明白。
所以为了更好的测试，我在原图上，画了道道，然后和 那一样的道道作为 mask。然后按照逻辑处理。就是如上第一种效果图。

**Comment by sunjunlishi** [comment]
按道理，两种方式验证效果应该一样。因为inpaint 不受背景影响，背景搞的本来就是好的，mask融合后，处理效果即使好，没有太大的意义，我测试 英伟达的那个 也是必须在 很好的原始上面画mask，搞张本来就是损坏的原图，英伟达的demo效果非常不好。

**Comment by zzj0311** [comment]
@sunjunlishi 训练集的输入必须是未经处理过的原图，否则训练出的模型性能是有问题的

**Comment by zzj0311** [comment]
如果我的理解有误的话，请给出具体的执行流程（代码），谢谢。

**Comment by sunjunlishi** [comment]
解释下：你直接 ps 在 好的图上 用画笔画几道，让程序修复看能修复不能。

**Comment by sunjunlishi** [comment]
用个好图测试没有什么太大的说服力，虽然你 乘以了 mask。但是程序内是否利用好的原图信息，我没深入程序。

**Comment by zzj0311** [comment]
我重新瞄了一眼，我的代码测试阶段并不允许自定义mask，请提供复现代码

**Comment by sunjunlishi** [comment]
1 简单的做法，是用 ps 你自己的mask 颜色在图上画上东西；  
   代码 celebA.py 中 __getitem__ 中  return gt*mask, mask, gt 前面的 gt*mask，你直接 返回 gt,让程序处理
2 random_ff_mask  随机生成的 mask，你可以搞个mask png 直接 imread
已经如此了。
问题我已经找到解决方案，pix2pix   https://github.com/GuoxiangQu/pix2pix-caffe。
unet网络。效果非常不错。

**Comment by zzj0311** [comment]
直接返回的gt有经过normarlize（-均值 再/方差）导致mask部分实际不为0，与训练时的统计特性有差异，而模型二三并不接收mask信息，因此性能衰退很正常，具体看report
